src/connectors/smart-sort/connectSmartSort.ts

```diff
@@ -0,0 +1,126 @@
+import { Connector } from '../../types';
+import {
+  createDocumentationMessageGenerator,
+  mergeSearchParameters,
+  noop,
+} from '../../lib/utils';
+import algoliasearchHelper from 'algoliasearch-helper';
+
+export type SmartSortConnectorParams = {
+  relevancyStrictness?: number;
+};
+
+export type SmartSortRendererOptions = {
+  isSmartSorted: boolean;
+  refine: (relevancyStrictness: number) => void;
+};
+
+const withUsage = createDocumentationMessageGenerator({
+  name: 'smartSort',
+  connector: true,
+});
+
+export type SmartSortConnector = Connector<
+  SmartSortRendererOptions,
+  SmartSortConnectorParams
+>;
+
+const connectSmartSort: SmartSortConnector = function connectSmartSort(
+  renderFn = noop,
+  unmountFn = noop
+) {
+  return widgetParams => {
+    if (
+      typeof widgetParams.relevancyStrictness !== 'undefined' &&
+      (widgetParams.relevancyStrictness < 0 ||
+        widgetParams.relevancyStrictness > 100)
+    ) {
+      throw new Error(
+        withUsage('The `relevancyStrictness` option must be between 0 and 100.')
+      );
+    }
+
+    return {
+      $$type: 'ais.smartSort',
+
+      init(initOptions) {
+        const { instantSearchInstance } = initOptions;
+        renderFn(
+          {
+            ...this.getWidgetRenderState(initOptions),
+            instantSearchInstance,
+          },
+          true
+        );
+      },
+
+      render(renderOptions) {
+        const { instantSearchInstance } = renderOptions;
+
+        renderFn(
+          {
+            ...this.getWidgetRenderState(renderOptions),
+            instantSearchInstance,
+          },
+          false
+        );
+      },
+
+      dispose({ state }) {
+        unmountFn();
+
+        return state.setQueryParameter('relevancyStrictness', undefined);
+      },
+
+      getRenderState(renderState, renderOptions) {
+        return {
+          ...renderState,
+          smartSort: this.getWidgetRenderState(renderOptions),
+        };
+      },
+
+      getWidgetRenderState({ results, helper }) {
+        const { appliedRelevancyStrictness } = results || {};
+
+        return {
+          isSmartSorted:
+            typeof appliedRelevancyStrictness !== 'undefined' &&
+            appliedRelevancyStrictness > 0 &&
+            appliedRelevancyStrictness <= 100,
+          refine: relevancyStrictness => {
+            helper
+              .setQueryParameter('relevancyStrictness', relevancyStrictness)
+              .search();
+          },
+          widgetParams,
+        };
+      },
+
+      getWidgetSearchParameters(state, { uiState }) {
+        return mergeSearchParameters(
+          new algoliasearchHelper.SearchParameters({
+            relevancyStrictness: widgetParams.relevancyStrictness,
```

Comment:
is this meant to be the default strictness? 

I think it's meant to toggle between:

1. undefined (100 set by the engine)
2. user-provided value (default to 0)

Reply:
I see two cases:

1. smartSort({ }); // no explicit relevancyStrictness
-> Clicking "See all results" set 0 to suppress it.
-> Clicking "See relevant results" set undefined to fallback to the value in the index configuration.

2. smartSort({ relevancyStrictness: xx });
-> Clicking "See all results" set 0 to suppress it.
-> Clicking "See relevant results" go back to the value in the widgetParams.

What do you think?